### PR TITLE
Extended logic eval error

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Return a bit more details on eval failure for app calls.

## Test Plan

Added `EvalContext.PcDetails()` method and tests.

### CODE REVIEW

**Review by jannotti — CHANGES_REQUESTED** [review]
I think that the pc is set to the end or maybe even 1 past the end for errors that don't get detected until after the first pass (when jumps are resolved).  Could you put in a test to show things work ok when PcDetails is called then?

**Review by jannotti — APPROVED** [review]
Seems pretty clear you would just get an empty string, so that seems ok.

File: ledger/appcow.go [hunk 1]
```diff
@@ -490,7 +490,9 @@ func (cb *roundCowState) StatefulEval(params logic.EvalParams, aidx basics.AppIn
 	var cx *logic.EvalContext
 	pass, cx, err = logic.EvalStatefulCx(program, params)
 	if err != nil {
-		return false, transactions.EvalDelta{}, ledgercore.LogicEvalError{Err: err}
+		pc, det := cx.PcDetails()
```
Comment (tsachiherman) [inline ledger/appcow.go:493]: This is fine as long as cx is non-nil. The default go pattern is that whenever err is non nil, all the other params are..
So, adding a nil check before working on cx wouldn't be a bad idea.

File: data/transactions/logic/eval.go [hunk 2]
```diff
@@ -3938,3 +3938,29 @@ func opTxSubmit(cx *EvalContext) {
 	})
 	cx.subtxn = nil
 }
+
+// PcDetails return PC and disassembled instructions at PC up to 2 opcodes back
+func (cx *EvalContext) PcDetails() (pc int, dis string) {
+	const maxNumAdditionalOpcodes = 2
+	text, ds, err := disassembleInstrumented(cx.program, nil)
+	if err == nil {
```
Comment (tsachiherman) [inline data/transactions/logic/eval.go:3946]: Nit: maybe an early return here ?

File: ledger/appcow.go (hunk 1 above)
Comment (algorandskiy) [reply ledger/appcow.go]: right, added the check.

File: data/transactions/logic/eval.go (hunk 2 above)
Comment (algorandskiy) [reply data/transactions/logic/eval.go]: done

**Comment by algorandskiy** [comment]
> Could you put in a test to show things work ok when PcDetails is called then?
added a test

**Review by tsachiherman — APPROVED** [review]
looks good.

**Pull request merged**